File: studio/services/profile_facts.py
"""Business profile fact extraction for generation services."""


def business_facts(profile):
    business = profile.get("business") or {}
    facts = []
    if business.get("type"):
        facts.append(f"- Business type: {business['type']}")
    if business.get("description"):
        facts.append(f"- Description: {business['description']}")
    if business.get("services"):
        facts.append(f"- Services: {', '.join(business['services'][:6])}")
    if business.get("targetAudience"):
        facts.append(f"- Target audience: {business['targetAudience']}")
    return "\n".join(facts)

File: src/server/profileFacts.ts
/** Business profile fact extraction for generation services (port of profile_facts.py). */

export function businessFacts(profile: Record<string, unknown>): string {
  const business = (profile.business as Record<string, unknown> | undefined) ?? {};
  const facts: string[] = [];
  if (business.type) facts.push(`- Business type: ${business.type}`);
  if (business.description) facts.push(`- Description: ${business.description}`);
  if (Array.isArray(business.services) && business.services.length) {
    facts.push(`- Services: ${business.services.slice(0, 6).join(", ")}`);
  }
  if (business.targetAudience) facts.push(`- Target audience: ${business.targetAudience}`);
  return facts.join("\n");
}
